### ISSUE TITLE
Throw and try/catch

### ISSUE THREAD

**Issue by mhulse** [mhulse/basic-node-module#2]
Use just throw and try/catch around awaits:

https://github.com/mhulse/basic-node-module/blob/5df8ced0c069a5dab1ba2bd609a3de977df4e6fd/index.js#L51-L65

Here's example of just throw:

https://github.com/mhulse/get-tile-url/blob/f79b35889e0ad43fda12053cd1d2a1dcd8ef3b61/index.js#L90

**Comment by mhulse** [comment]
https://github.com/i0natan/nodebestpractices#-22-use-only-the-built-in-error-object

**Comment by mhulse** [comment]
Great examples here:

https://github.com/sindresorhus/p-queue/blob/37c1204cc68e54f9d4ecf4e5200eb9313fd277be/index.js#L69-L79

**Comment by mhulse** [comment]
Good advice here: https://stackoverflow.com/a/52215925/922323

**Comment by mhulse** [comment]
This talks about how to use top-level caller:

https://stackoverflow.com/a/48685979/922323

**Comment by mhulse** [comment]
Sync way: https://stackoverflow.com/questions/30367395/how-to-catch-errors-in-synchronous-functions-in-node-js

**Comment by mhulse** [comment]
This is interesting:

https://makandracards.com/makandra/43653-javascript-don-t-throw-exceptions-from-async-functions

> TLDR: A function is hard to use when it sometimes returns a promise and sometimes throws an exception. When writing an async function, prefer to signal failure by returning a rejected promise.

**Comment by mhulse** [comment]
I need to make a readme wiki repo on exception handling, just for my own education. This shit is probably one of the more complex/ambiguous feature(?) of es6+ (in my mind).

Just need more practice. :)

**Comment by mhulse** [comment]
Looks like thowing a new error will result in a rejected promise in a async/await.

https://stackoverflow.com/a/42453705/922323

**Comment by mhulse** [comment]
A few good paragraphs about constructors and async operations; basically, don't do that in constructor.

**Comment by mhulse** [comment]
Great tips here:

https://eloquentjavascript.net/08_error.html

> Another way to set a breakpoint is to include a debugger statement (consisting of simply that keyword) in your program. If the developer tools of your browser are active, the program will pause whenever it reaches such a statement.

> If you’re programming only for yourself, you can afford to just ignore such problems until they occur. But if you build something that is going to be used by anybody else, you usually want the program to do better than just crash. Sometimes the right thing to do is take the bad input in stride and continue running. In other cases, it is better to report to the user what went wrong and then give up. But in either situation, the program has to actively do something in response to the problem.

**Comment by mhulse** [comment]
https://medium.com/@xjamundx/custom-javascript-errors-in-es6-aa891b173f87

**Comment by mhulse** [comment]
Good tips here:

https://stackoverflow.com/a/42171508/922323

**Comment by mhulse** [comment]
Returning a promise from async will not double wrap the promise:

https://stackoverflow.com/a/35302535/922323

Non-promise return will be wrapped.

No return will return a promise.
